ha/core/node/node_manager.py

```diff
@@ -0,0 +1,174 @@
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+from cortx.utils.ha.dm.decision_monitor import DecisionMonitor
+from cortx.utils.log import Log
+from cortx.utils.schema.conf import Conf
+
+from ha import const
+from ha.core.node.node import Node
+from ha.core.node.replacement.refresh_context import CortxRefreshContext
+from ha.core.service.service_manager import CortxServiceManager
+from ha.core.error import HAUnimplemented
+
+
+class NodeManager:
+    """
+    Node Manager
+    """
+    def __init__(self):
+        """
+        Manage node operation
+        """
+        pass
+
+    def process_request(self, action, args, output):
+        """
+        Generic method to handle process request
+
+        Args:
+            action ([string]): Take cluster action for each request.
+            args ([dict]): Parameter pass to request to process.
+
+        Raises:
+            HAUnimplemented: Unimplemented method.
+        """
+        raise HAUnimplemented()
+
+    def get_node_instance_list(self):
+        """
+        Returns node list
+        """
+        raise HAUnimplemented()
+
+    def get_node_instance(self, node_id):
+        """
+        Returns instance of the node
+        """
+        raise HAUnimplemented()
+
+    def validate_node(self, node_id):
+        """
+        Validate requested node is there in the node list
+        """
+        raise HAUnimplemented()
+
+class CortxNodeManager(NodeManager):
+    def __init__(self):
+        """
+        Prepare node instance list
+        """
+        self.node_instance_list = {}
+        self._service_manager = CortxServiceManager()
+
+        node_list = Conf.get(const.HA_GLOBAL_INDEX, "Node.nodelist")
+
+        for node_id in node_list.keys():
+            self.node_instance_list[node_id] = Node(node_id)
+
+        Log.debug(f"node_instance_list {self.node_instance_list}")
+
+        self._decision_monitor = DecisionMonitor()
+        self._refresh_context = CortxRefreshContext(self._decision_monitor)
+
+    def process_request(self, action, args, output):
+        """
+        Generic method to handle process request
+
+        Args:
+            args ([dict]): Parameter pass to request to process.
+        """
+        if action == const.NODE_COMMAND:
```

Comment:
const.NODE_COMMAND:
for main node action is raise HAUnimplemented()

Reply:
Why it should be unimplemented? Could you please elaborate more?